File: client/rn-arkit-replacement/modules/LiDAR/ios/Sources/Native/RNLiDAREventEmitter.m
#import <React/RCTEventEmitter.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNLiDAREventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(sendMeshUpdate:(NSDictionary *)data)

@end

File: client/rn-arkit-replacement/ios/UniWhereLiDAR/Native/RNLiDAREventEmitter.swift
import Foundation
import React

@objc(RNLiDAREventEmitter)
final class RNLiDAREventEmitter: RCTEventEmitter {

    override func supportedEvents() -> [String]! {
        return ["onMeshUpdate"]
    }

    @objc
    func sendMeshUpdate(_ data: [String: Any]) {
        sendEvent(withName: "onMeshUpdate", body: data)
    }

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
